Platform/Authentication/Error Handling/AuthenticationError.swift

```diff
@@ -0,0 +1,15 @@
+import Foundation
+
+/// Enumeration that represents authentication error.
+public enum AuthenticationError: Error {
```

Comment:
Do we expect clients of `DefaultTokenProvider` to handle the different case's of `AuthenticationError`?   

Because I was wondering if `AuthenticationResult` should be more strongly typed, i.e. `case failure(error: AuthenticationError)` which would more easily allow for this:

```
        tokenProvider.fetchToken { result in
            switch result {
            case let .authenticated(token):
                // All good
            case let .failure(authenticationError: AuthenticationError):
                switch authenticationError {
                case .invalidURL:
                    // Do something specific
                case .failedToSerializeBody:
                    // Do something specific
                case .failedToParseToken:
                    // Do something specific
                case let .unexpected(error: Error):
                    // Do something specific
                }
            }
        }
```

We could even let clients define their own generic `Error` on a `TokenProvider` with `DefaultTokenProvider` using `AuthenticationError`?

I'm quite probably over engineering this though :-)

Reply:
I added additional logging.

Reply:
👍